aws-ia/terraform-aws-eks-blueprints
# docs: Update and standardize examples to follow conventions defined in pre-commit checks

### What does this PR do?
- Standardize AWS provider region usage within examples; remove use of variable for region in provider since this should be static
- Remove references to prior license material and MIT licensing
- Standardize provider and required versions within `versions.tf` per conventions; this also aligns versions that are used to be consistent
- Update pre-commit checks
	- Update versions to use latest
	- Remove `.tflint.hcl` file that wasn't enforcing lint rules, add rules to pre-commit to ensure conventions are enforced
	- Update `terraform_docs` to ensure lockfile doesn't affect version shown in READMEs
- Update and align version used on VPC module within examples
- Move outputs to `outputs.tf` file per convention; add empty file where one is not present in the directory
- Correct tflint errors
	- Remove  unused locals and data sources
	- Ensure variables and outputs have descriptions
	- Ensure names use underscores instead of hyphens
	- The managed node group policy attachment resource names are left as is with a `TODO` for now since that is a breaking change  
	- Ensure provider versions are set in required providers
	- Quite a few whitepsace changes
- Correct `terraform_docs` replacement pattern so that pre-commit will update documentation on detected changes; this was not updating previously and the documentation was out of sync with the source
- Update `kubernetes`, `helm`, and `kubectl` providers to use `exec()` method per the Terraform provider docs. This follows same pattern used by the EKS service within the `kubeconfig` file to ensure credentials are refreshed. This also removes the use of two data sources `aws_eks_cluster_auth` and `aws_eks_cluster` per example

### Motivation
- Standardize and align with current conventions set within static analysis tools
- All pre-commit checks pass successfully
- Closes #469 

### More

- [ ] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [x] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR


**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [ ] E2E Test successfully complete before merge?

### Additional Notes
